### ISSUE TITLE
bcftools consensus not calling all ambiguities.

### ISSUE THREAD

**Issue by sirswanton** [samtools/bcftools#1703]
This is somewhat following on a previous post (https://github.com/samtools/bcftools/issues/1516).

Briefly, after running bcftools consensus -I, I noticed sites were being called as ambiguous (Y) where the REF was C but all the samples were homozygous for T. 

After reading about this issue here, I changed to bcftools consensus -H I -s samples, which fixed this problem.

However, now I noticed that some legitimate dimorphic alleles are being called as homozygous rather than ambiguous. These have low (~0.04-0.06) AF, but adding a flag -i 'INFO/AF >= 0.03' to try and force these does not change anything. 

One that was correctly called had a higher AF (0.12). I manually changed the AF of the others to this value, but nothing changed. Is there a reason I'm missing why these dimorphic alleles, which were called using 'bcftools consensus -I' are not called now that I switched to 'bcftools consensus -H I -s samples'? 

Basically, I am wondering how exactly bcftools consensus decides which base to write to the fasta. 

Is the AF score taken from INFO/AF or calculated from the GT for each sample? Are GQ or any other parameters used to weight the value of a variant, or are they all treated equally? Is the default "bcftools consensus -H I" command to write every base with at least one heterozygote as ambiguous?

**Comment by pd3** [comment]
In order to keep the program simple, the `consensus` command merely applies ALL variants from the VCF regardless of their AF or other properties. If more more advanced logic is required, the VCF has to be preprocessed with other tools, such as the `+setGT` plugin.
